### PR TITLE
[v2-rc] teardown old v1 ansible local fabrics when upgrading to different major version

### LINKED ISSUE #2872: [v2 merge] V1 migration docker cleanup
## Description

If a user is migrating from `v1.0.40` (or an older v1 version) we should do Docker container, volume & network cleanup.
We could either do this in the background when the user installs v2, or ask the user if they would like us to do cleanup before continuing.
## Steps to Reproduce

1.
2.
3.

## Existing issues

- [ ] [Stack Overflow issues](https://stackoverflow.com/questions/tagged/ibp-vscode-extension)
- [ ] [GitHub Issues](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues)

## Your Environment

* IBM Blockchain Platform Extension Version:
* Visual Studio Code version:
* Environment name and version (e.g. Node.js v8.12.0, npm v6.4.1):
* Operating System and version:
* Link to your project:

### PR DESCRIPTION
**Pull request by lquintai**
Closes #2872 

Signed-off-by: Leonor Quintais <lquintai@uk.ibm.com>

### CODE REVIEW

File: packages/blockchain-extension/extension/fabric/environments/EnvironmentFactory.ts [hunk 1]
```diff
@@ -33,9 +33,10 @@ export class EnvironmentFactory {
         }
 
         const type: EnvironmentType = environmentRegistryEntry.environmentType;
+        const isV1LocalAnsible: boolean = type === EnvironmentType.MANAGED && environmentRegistryEntry.environmentDirectory && environmentRegistryEntry.environmentDirectory.indexOf('/v2/environments/') === -1;
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/fabric/environments/EnvironmentFactory.ts:36]: This path wouldn't be valid on Windows.

File: packages/blockchain-extension/test/fabric/environments/EnvironmentFactory.test.ts [hunk 2]
```diff
@@ -110,4 +111,17 @@ describe('EnvironmentFactory', () => {
         environment.should.be.an.instanceOf(MicrofabEnvironment);
     });
 
+    it(`should return a local environment when old v1 ansible environment`, async () => {
+        // await TestUtil.setupLocalFabric();
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/test/fabric/environments/EnvironmentFactory.test.ts:115]: Can this comment be removed?

File: packages/blockchain-extension/extension/extension.ts [hunk 3]
```diff
@@ -116,8 +124,30 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
-
+        context = await ExtensionUtil.registerPreActivationCommands(context);
+
+        if (extensionUpgraded) {
+            // If user upgraded major versions, if old local ansible environmens present ask if they should be teardown
+            // This needs to happen before rendering pannels, otherwise the user will see the old environments
+            const v1ExtDir: string = extDir.replace(`${path.sep}v2`, '');
+            FabricEnvironmentRegistry.instance().setRegistryPath(v1ExtDir);
+            try {
+                const v1EnvironmentEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll();
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/extension.ts:135]: This returns ALL environments including environments which we've added with nodes, so they might be elsewhere.

File: packages/blockchain-extension/extension/extension.ts [hunk 4]
```diff
@@ -116,8 +124,30 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
+                if (v1EnvironmentEntries.length > 0) {
+                    const teardown: any = await UserInputUtil.showConfirmationWarningMessage('Detected old local environments which can not be used with this upgrade. Would you like to teardown these environments?');
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/extension.ts:137]: Type please!

File: packages/blockchain-extension/extension/extension.ts [hunk 5]
```diff
@@ -116,8 +124,30 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
+                    if (teardown) {
+                        for (const entry of v1EnvironmentEntries) {
+                            await vscode.commands.executeCommand(ExtensionCommands.TEARDOWN_FABRIC, undefined, true, entry.name, true);
```
Comment (Jakeeyturner) [inline packages/blockchain-extension/extension/extension.ts:140]: I think we should try/catch each teardown (can ignore the error), just so this won't potentially stop the entire extension.

Comment (lquintai) [reply packages/blockchain-extension/extension/extension.ts]: It wouldn't stop the extension, but I'm happy to try/catch each.

**Review by Jakeeyturner — APPROVED** [review]

**Pull request merged**